mobile-sdk/src/main/java/com/github/itwin/mobilesdk/ITMApplication.kt

```diff
@@ -185,25 +191,31 @@ open class ITMApplication(
     var frontendBaseUrl = ""
 
     /**
-     * The [ITMMessenger] for communication between native code and JavaScript code (and vice versa).
+     * The [ITMLogger] responsible for handling log messages (both from native code and JavaScript
+     * code). The default logger uses [Log][android.util.Log] for the messages. Replace this object
+     * with an [ITMLogger] subclass to change the logging behavior.
+     *
+     * > __Note:__ Updating this value also updates the logger being used by [messenger].
      */
-    @Suppress("LeakingThis")
-    var messenger = ITMMessenger(this)
+    var logger: ITMLogger by Delegates.observable(ITMLogger()) { _, _, new ->
+        messenger.logger = new
+    }
```

Comment:
Does observable provide any advantage over using a property setter? 

```suggestion
    var logger = ITMLogger()
        set (value) {
            field = value
            messenger.logger = value
        }
```

Reply:
Not that I know of, but the same question could be asked in reverse, and the observable is a bit more succinct.

Reply:
I don't mind changing it to a set if you really feel that is better, but I personally don't see either one as being better or worse.

Reply:
I don't have a strong opinion either way. Using a setter is more common I think. I'd never heard of observable before this.

Reply:
I only learned about observable when I was investigating lazy, but felt that it was appropriate here. Given no strong feeling either way, I'm going to leave this as it is.